docs/contributing/testing-guidelines/integration-testing.md

```diff
@@ -0,0 +1,253 @@
+    In such cases use the manual solution from the [component test section below](#integration-test-with-a-single-node-component-test).
+
+### Integration test with a single node: component test
+
+The simplest scenario is a single node. In this case, the integration test is commonly referred to as a component test.
+
+To add a component test to an existing node, follow the example of the `lanelet2_map_provider` package that has an executable named `lanelet2_map_provider_exe`.
+
+In `package.xml`, add:
+
+```xml
+<test_depend>ros_testing</test_depend>
+```
+
+In `CMakeLists.txt`, add or modify the `BUILD_TESTING` section:
+
+```cmake
+if(BUILD_TESTING)
+  find_package(ament_lint_auto REQUIRED)
+  ament_lint_auto_find_test_dependencies()
+
+  add_ros_test(
+    test/lanelet2_map_provider_launch.test.py
+    TIMEOUT "30"
+  )
+endif()
+```
+
+!!! note
+
+    The `TIMEOUT` argument is given in seconds; see the [add_ros_test.cmake file](https://github.com/ros2/ros_testing/blob/master/ros_testing/cmake/add_ros_test.cmake) for details.
+
+To create a test, either read the [launch_testing quick-start example](https://github.com/ros2/launch/tree/master/launch_testing#quick-start-example), or follow the steps below.
+
+Taking `test/lanelet2_map_provider_launch.test.py` as an example, first dependencies are imported:
+
+```python
+from ament_index_python import get_package_share_directory
+from launch import LaunchDescription
+from launch_ros.actions import Node
+import launch_testing
+
+import os
+import pytest
+import unittest
+```
+
+Then a launch description is created to launch the node under test. Note that the `test_map.osm` file path is found and passed to the node, something that cannot be done with the [smoke testing API](#smoke-tests):
+
+```python
+@pytest.mark.launch_test
+def generate_test_description():
+
+    map_osm_file = os.path.join(
+        get_package_share_directory('lanelet2_map_provider'),
+        'data/test_map.osm'
+    )
+
+    lanelet2_map_provider = Node(
+        package='lanelet2_map_provider',
+        executable='lanelet2_map_provider_exe',
+        namespace='had_maps',
+        parameters=[
+            os.path.join(
+                get_package_share_directory('lanelet2_map_provider'),
+                'param/test.param.yaml'
+            ),
+            {
+                'map_osm_file': map_osm_file
+            }]
+    )
+
+    context = {'lanelet2_map_provider': lanelet2_map_provider}
+
+    return LaunchDescription([
+        lanelet2_map_provider,
+        # Start tests right away - no need to wait for anything
+        launch_testing.actions.ReadyToTest()]
+    ), context
+```
+
+Finally the test condition. As before, it is just a smoke test that ensures the node can be
+
+1. Launched with its default parameter file.
+2. Terminated with a standard `SIGTERM` signal.
+
+So the test code is executed after the node executable has been shut down (`post_shutdown_test`):
```

Comment:
The description of the test condition mentions that we are testing that the node can be launched with its default parameter file and then terminated with a `SIGTERM` signal. However, the code below (`test_exit_code`) seems to be only testing the termination part and not the launching part. 

Should that also have been included?